File: training/java/core/Daily Task/Linked List insertion/src/LinkedList.java

public class LinkedList {
	public int val;
	public LinkedList next;
	public LinkedList(int val) {
		this.val=val;
		this.next=null;
	}
	public void add(int x,LinkedList head) {
		if(head==null) {
			head=new LinkedList(x);
		}
		else {
			LinkedList p = head;
			head.next=new LinkedList(x);
		}
	}
}

File: python/basics/Linkedlist.py
class Node:
    def __init__(self, val=0):
        self.val = val
        self.next = None


class Linkedlist:
    def __init__(self):
        self.head = None

    def add(self, val):
        if self.head == None:
            self.head = Node(val)
            return
        a = self.head
        while a.next:
            a = a.next
        a.next = Node(val)

    def display(self):
        temp = self.head
        while temp:
            print(temp.val)
            temp = temp.next


a = Linkedlist()
a.add(10)
a.add(20)
a.add(30)
a.display()
